### PR TITLE
Replace NEWS with NEWS.md with more details and examples

### PR DESCRIPTION
**Pull request by wader**
Changes mentioned based on picking user facing changes from: git log --oneline -r master...jq-1.6 | grep -v Merge

### CODE REVIEW

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Allow `if` without `else`-branch. When skipped the `else`-branch will be `.` (identity). 4f6045a94e0519ec58210bbd58ece442c32468e1 @chancez
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
  # previously parentheses were needed
```

**Review by jsoref — CHANGES_REQUESTED** [review]

File: NEWS.md
Comment (jsoref) [inline NEWS.md]: This probably shouldn't be mentioned? #2535

Comment (wader) [reply NEWS.md]: This one is a bit confusing, #1271 was not been merged so jq master seems to still have it. Also the shorthand `-0` does not work as jq's argument parser reads it as a negative number i think.

Comment (jsoref) [reply NEWS.md]: If `-0` doesn't work, then it shouldn't be documented. And if you're likely to remove the other flag, then even if it technically exists, I'd suggest not announcing it, at least, without some big hazard warnings that it's likely to be removed and pointing to an alternative.

Comment (wader) [reply NEWS.md]: Now i remember. Short flag `-0` do work but only if used in combination with other short flags like in `-n0`. This is very confusing, so maybe we should only use long flag for this or come up with a different short flag? or fix the argument parsing code somehow?

I do think null separated output can be useful (but can be dangerous with `-r`) so personally i would like to keep it in some form.

@itchyny what do you think? and i also noticed gojq has no issue with this:
```sh
$ gojq -n -0 123
123% # (% is zsh showing there is no new line)
$ jq -n -0 123
-0
```

Comment (jsoref) [reply NEWS.md]: Wow, that's indeed unfortunate.

In the interim, I'd definitely suggest not documenting it, and maybe even removing it until the behavior can be implemented in a less confusing manner.

I personally am a fairly heavy `-z` / `-0` user of various things (e.g. `git ls-files -z|xargs -0` or `find . -print0|xargs -0` or things where my own code consumes/produces `nul` delimited content).

Comment (wader) [reply NEWS.md]: Seem like @itchyny found and old PR to fix the argument parsing https://github.com/jqlang/jq/pull/2235

Comment (itchyny) [reply NEWS.md]: Yeah, I merged the PR so now we can use `-0` with the master version.
```bash
 $ jq --version
jq-1.6-180-g6c5c8f2

 $ jq -n -0 1,2,3 | xargs -0 -n1
1
2
3
```

Comment (jsoref) [reply NEWS.md]: Do you want to change the PR reference to also include that PR?

Comment (itchyny) [reply NEWS.md]: Yes, that's better to be included in this list.

Comment (wader) [reply NEWS.md]: Yes will fix and also steal @itchyny example once im at home

Comment (itchyny) [reply NEWS.md]: Using short flag combination is user's preference, so I think it's better not to use it in the release note. Maybe more example which cannot be archived before would help user know how it is useful?
```sh
$ jq -n -0 '1,2,3' | xxd
00000000: 3100 3200 3300                           1.2.3.
# can be used with xargs -0
$ jq -n -0 '"a b c", "d\ne\nf"' | xargs -0 printf '%q\n'
'a b c'
'd'$'\n''e'$'\n''f'
# can be used with read -d ''
$ while IFS= read -r -d '' json; do
>   jq '.name' <<< "$json"
> done < <(jq -n -0 '{name:"a b c"},{name:"d\ne\nf"}')
"a b c"
"d\ne\nf"
```

Comment (wader) [reply NEWS.md]: Good idea, added your examples, changed my 1,2,3 examples to us a,b,c instead, seems better. Also changed to correct say that zero byte is after each output not between.

Could you clarify what you mean by "Using short flag combination is user's preference, so I think it's better not to use it in the release note"? use `-n -0`  not `-n0` in  release notes?

**Comment by itchyny** [comment]
Maybe we add NEWS.md to `dist_doc_DATA` even if newer autotools includes this in the dist file? Ref: #2482.

**Comment by wader** [comment]
@itchyny 👍sorry for being a bit slow, on vacation, but will probably have some jq time later in the week

**Comment by pkoppstein** [comment]
@wader - Looking at the current NEWS.md ...

I think it would be important to organize the news carefully, e.g. along the following lines,
preferably with a TOC and headers:

1) news about jq command-line options

2) changes that might affect the behavior of existing jq programs 
    a) notable bug fixes
    b) numeric literals
    c) removal of (deprecated) filters
    d) stderr: removal of decorations
    ...
    
3) language enhancements
   * else is optional
   * destructuring
   * keywords can be used in $-variable names
   ...
   
3) new builtins
   a) builtins to support binary strings
   b) debugging enhancement
   c) abs, ...
   ...

4) performance enhancements
    (I wouldn't mention run-of-the-mill enhancements such as to `transpose`)

5) release builds

6) other news
   a) quality assurance
   b) web site

7) Acknowledgements

---
ALSO:

* Please note that one of the main reasons for `abs` is that it won't force the conversion
to floats that length and fabs actually entail.

**Comment by wader** [comment]
@pkoppstein Thanks, all good input and will update about abs and maybe remove about transpose. You can also add a reviews to suggested wording etc if you want

Yeap some more structure might be good but i also don't think it should not be too structured. I've also tried to sort things in order of "news worthyness for users". Anyway now when most changes with author/PRs etc is collect it's quite quick to restructure things.

File: NEWS.md
Comment (wader) [inline NEWS.md]: Was thinking maybe nice to start with some kind summary/introduction like this? what do ppl think? more/less formal?

**Comment by wader** [comment]
My idea was also to be able to use the markdown for the 1.7 section more or less as-is for the release on github.

**Comment by wader** [comment]
Should we mention https://github.com/jqlang/jq/pull/2750? seems to have quite a lot of dup issues related to it

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]: How about "New maintainers, admins, and owners have been recruited."?

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]: s/Rename `--nul-output` to/Add/ (since it never shipped)

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]: s/Langauge/Language/

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]: Maybe this should be higher up?  A lot of people want this.

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- A new team of maintainers has been organized.
```

(Optionally including a parenthetical clarifying that some are new and some are old).

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- CI, scan builds, release builds etc now use GitHub actions.
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Improved `--help` output. Now mentions all options and nicer order. @itchyny #2747 #2766
```

File: NEWS.md
Comment (jsoref) [reply NEWS.md]: Sigh. check-spelling would catch this 😉

**Comment by nicowilliams** [comment]
Thank you so much for this beautiful NEWS file!  I've left a few comments, but it LGTM modulo those comments.

File: NEWS.md
Comment (wader) [reply NEWS.md]: Yeap better, changing to that

File: NEWS.md
Comment (wader) [reply NEWS.md]: Noticed now that it actually did but i didn't catch it :)

File: NEWS.md
Comment (wader) [reply NEWS.md]: Moved first. I guess we want to group all new builtins?

File: NEWS.md
Comment (wader) [reply NEWS.md]: Changed to @nicowilliams suggestion "New maintainers, admins, and owners have been recruited.". What do you think @nicowilliams?

**Comment by wader** [comment]
Thanks @jsoref!

**Comment by wader** [comment]
> Thank you so much for this beautiful NEWS file! I've left a few comments, but it LGTM modulo those comments.

Thanks, and i guess we can leave this PR open for now and collect comments

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]:
```suggestion
Lots of JSON has been queried and transformed since the last stable release, jq-1.6, 5 years ago, so now we're happy to finally do a new stable release!
```

File: NEWS.md
Comment (nicowilliams) [inline NEWS.md]: We should probably mention the try/catch (#1859) bug fix as well.  That one had lots of dup issues.

File: NEWS.md
Comment (nicowilliams) [reply NEWS.md]: Maybe a bit less formal, yeah.  "After a five year hiatus we're back with a GitHub organization, with new admins and new maintainers who have brought a great deal of energy to make a long-awaited and long-needed new release."?

Comment (wader) [reply NEWS.md]: Yeap better

File: NEWS.md
Comment (jsoref) [inline NEWS.md]: Did you mean etcetera? -- https://github.com/check-spelling/jq/pull/3/files#diff-51920e95310ebfbc1ae31709f3b95f89afffbf4f1a6e38e8b2b406e2fb6197eaR298

File: NEWS.md
Comment (jsoref) [inline NEWS.md]: Could I make a plea for consistency?

```
@itchyny #2728
```
And
```
#1697 @ryo1kato
```

If you used a single style, I could write a pattern to ignore them, and it'd be easier for normal readers to have a single pattern to see...

Comment (wader) [reply NEWS.md]: It's suppose to be in one style @user @user... #pr #pr... so is an error, thanks! Was thinking @user #pr @user #pr for a while but sometimes more ppl have been involved then just the PR author so i went with collection of users involed then collection of related PRs

**Comment by wader** [comment]
It would be nice if the mentions of builtins like `abs/0` could be links to the documentation? haven't checked if possible, maybe would also require links to be absolute, maybe not good?

**Comment by nicowilliams** [comment]
Any objections to me merging this right now?

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Allow `if` without `else`-branch. When skipped the `else`-branch will be `.` (identity). @chancez @wader #1825 #2481
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Fix issue converting string to number after previous convert error. @thalman #2400
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Use decimal number literals to preserve precision. Comparison operations respects precision but arithmetic operations might truncate. @leonid-s-usov #1752
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Allow dot between chained indexes when using `.["index"]` @nicowilliams #1168
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md]:
```suggestion
- Speed up and refactor some builtins, also remove `scalars_or_empty/0`. @muhmuhten #1845
```

File: NEWS.md
Comment (jsoref) [inline NEWS.md:121]:
```suggestion
- Adds new builtin [`abs`](https://jqlang.github.io/jq/manual/#abs) to get absolute value. This was previously possibly using `length` or `fabs` but naming was a bit confusing. @pkoppstein #2767
```

**Review by jsoref — COMMENTED** [review]
Documentation links?

File: NEWS.md
Comment (wader) [reply NEWS.md]: Is mention as `- Fix try/catch catches more than it should. @nicowilliams #2750` now, is good?

File: NEWS.md
Comment (wader) [reply NEWS.md:121]: @nicowilliams @itchyny  what do you think? links?

**Comment by wader** [comment]
@jsoref Thanks again, not sure how i got some many of them wrong 🤔

**Comment by wader** [comment]
> Any objections to me merging this right now?

I'm ok with merging. Only thing is if we want documentation links?

**Comment by nicowilliams** [comment]
> Is mention as `- Fix try/catch catches more than it should. @nicowilliams #2750` now, is good?

Yes.

**Comment by nicowilliams** [comment]
> I'm ok with merging. Only thing is if we want documentation links?

We can always do that in a separate PR :)

Thanks!

**Pull request merged**
